src/Components/Web.JS/src/Rendering/Events/EventTypes.ts

```diff
@@ -0,0 +1,377 @@
+    if (aliasGroup) {
+      aliasGroup.push(eventName);
+    } else {
+      browserEventNamesToAliases.set(options.browserEventName, [eventName]);
+    }
+
+    // For developer convenience, it's allowed to register the custom event type *after*
+    // some listeners for it are already present. Once the event name alias gets registered,
+    // we have to notify any existing event delegators so they can update their delegated
+    // events list.
+    eventNameAliasRegisteredCallbacks.forEach(callback => callback(eventName, options.browserEventName));
+  }
+
+  eventTypeRegistry.set(eventName, options);
+}
+
+export function getEventTypeOptions(eventName: string): EventTypeOptions | undefined {
+  return eventTypeRegistry.get(eventName);
+}
+
+export function getEventNameAliases(eventName: string): string[] | undefined {
+  return browserEventNamesToAliases.get(eventName);
+}
+
+export function getBrowserEventName(possibleAliasEventName: string): string {
+  const eventOptions = eventTypeRegistry.get(possibleAliasEventName);
+  return eventOptions?.browserEventName || possibleAliasEventName;
+}
+
+function registerBuiltInEventType(eventNames: string[], options: EventTypeOptions) {
+  eventNames.forEach(eventName => eventTypeRegistry.set(eventName, options));
+}
+
+registerBuiltInEventType(['input', 'change'], {
+  createEventArgs: parseChangeEvent
+});
+
+registerBuiltInEventType(['copy', 'cut', 'paste'], createBlankEventArgsOptions);
+
+registerBuiltInEventType(['drag', 'dragend', 'dragenter', 'dragleave', 'dragover', 'dragstart', 'drop'], {
+  createEventArgs: e => parseDragEvent(e as DragEvent)
+});
+
+registerBuiltInEventType(['focus', 'blur', 'focusin', 'focusout'], createBlankEventArgsOptions);
+
+registerBuiltInEventType(['keydown', 'keyup', 'keypress'], {
+  createEventArgs: e => parseKeyboardEvent(e as KeyboardEvent)
+});
+
+registerBuiltInEventType(['contextmenu', 'click', 'mouseover', 'mouseout', 'mousemove', 'mousedown', 'mouseup', 'dblclick'], {
+  createEventArgs: e => parseMouseEvent(e as MouseEvent)
+});
+
+registerBuiltInEventType(['error'], {
+  createEventArgs: e => parseErrorEvent(e as ErrorEvent)
+});
+
+registerBuiltInEventType(['loadstart', 'timeout', 'abort', 'load', 'loadend', 'progress'], {
+  createEventArgs: e => parseProgressEvent(e as ProgressEvent)
+});
+
+registerBuiltInEventType(['touchcancel', 'touchend', 'touchmove', 'touchenter', 'touchleave', 'touchstart'], {
+  createEventArgs: e => parseTouchEvent(e as TouchEvent)
+});
+
+registerBuiltInEventType(['gotpointercapture', 'lostpointercapture', 'pointercancel', 'pointerdown', 'pointerenter', 'pointerleave', 'pointermove', 'pointerout', 'pointerover', 'pointerup'], {
+  createEventArgs: e => parsePointerEvent(e as PointerEvent)
+});
+
+registerBuiltInEventType(['wheel', 'mousewheel'], {
+  createEventArgs: e => parseWheelEvent(e as WheelEvent)
+});
+
+registerBuiltInEventType(['toggle'], createBlankEventArgsOptions);
```

Comment:
All the logic for built-in event types here is just a reformulated version of what we used to have in `eventForDotNet.ts` before. The built-in event types are held in the same data structure we use for custom event registrations, so it's easy to treat them all equivalently when processing events, and we can guarantee uniqueness of names.